dds/DCPS/transport/tcp/TcpConnection.cpp

```diff
@@ -112,18 +131,17 @@ OpenDDS::DCPS::TcpConnection::disconnect()
 {
   DBG_ENTRY_LVL("TcpConnection","disconnect",6);
   this->connected_ = false;
-  TcpReceiveStrategy_rch receive_strategy = this->receive_strategy();
-  if (receive_strategy) {
-    receive_strategy->get_reactor()->remove_handler(this,
-                                                    ACE_Event_Handler::READ_MASK | ACE_Event_Handler::DONT_CALL);
+
+  if (interceptor_) {
+    ReactorInterceptor::CommandPtr cmd = interceptor_->execute_or_enqueue(new RemoveHandler(RcHandle<TcpConnection>(this, inc_count()), READ_MASK | DONT_CALL));
```

Comment:
There's a lot going on here.  Maybe break out the `TcpConnection_rch` construction to its own automatic variable. (same with other enqueue_or_execute calls below)